src/Weather.java

```diff
@@ -0,0 +1,170 @@
+     * Kernel methods ---------------------------------------------------------
+     */
+    /**
+     * Adds the pair ({@code date}, {@code condition}) to this.
+     *
+     * @param key
+     *            the key to be added
+     * @param value
+     *            the associated value to be added
+     * @aliases references {@code key, value}
+     * @updates this
+     * @requires key is not in DOMAIN(this)
+     * @ensures this = #this union {(key, value)}
+     */
+    public final void addWeather(String date, String condition) {
+        assert date != null : "Violation of: date is not null";
+        assert condition != null : "Violation of: condition is not null";
+        assert this
+                .isDateFormatted(date) : "Violation of: date is not formatted";
+        this.weather.add(date, condition);
+    }
+
+    public final Weather<date, condition> removeWeather(String date) {
+        assert date != null : "Violation of: date is not null";
+        assert this.isDateFormatted(date) : "Violation of: date is formatted";
+        assert this.hasDate(date) : "Violation of: date is in this";
+        Map.Pair<String, String> pair = this.weather.remove(date);
+        return new Weather<>(pair.key(), pair.value());
+    }
+
+    public final boolean hasDate(String date) {
+        assert date != null : "Violation of: date is not null";
+        assert this.isDateFormatted(date) : "Violation of: date is formatted";
+        return this.weather.hasKey(date);
+    }
+
+    /*
+     * Ideal date format: "YYYY-MM-DD"
+     */
+    public final boolean isDateFormatted(String date) {
+        assert date != null : "Violation of: date is not null";
+        boolean result = date.length() == 10;
+        if (result) {
+            // Check if format is appropriate
+            String space = date.substring(4, 5) + date.substring(7, 8);
+            if (space.equals("--")) {
+                String year = date.substring(0, 4);
+                // Check if year is appropriate number
+                if (FormatChecker.canParseInt(year)) {
+                    String month = date.substring(5, 7);
+                    // Check if month is appropriate number
+                    if (FormatChecker.canParseInt(month)) {
+                        int MM = Integer.parseInt(month);
+                        result = 1 <= MM && MM <= 12; // in range from 1 to 12
+                        String day = date.substring(8, 10);
+                        // Check if day is appropriate number
+                        if (result && FormatChecker.canParseInt(day)) {
+                            int DD = Integer.parseInt(day);
+                            /*
+                             * Not checking if day is truly apporiate for this
+                             * minimum viable product e.g. 02/31 is not possible
+                             */
+                            result = 1 <= DD && DD <= 31; // in range from 1 to 31
+                        } else {
```

Comment:
If you initially set result to false, you probably don't need these else statements.